docs/src/proposals/vote_subcommittee.md

```diff
@@ -0,0 +1,137 @@
+
+
+* `slow_hash`: repeated sha256 for rounds equal to 1ms on modern
+sha-ni hardware.
+
+### Safety Violations
+
+An Optimistic Confirmation violation occurs if a voting subcommittee
+optimistically confirms a slot that wasn't rooted by the following
+voting subcommittee.
+
+A Safety violation occurs if a subcommittee roots a slot that wasn't
+rooted by the following subcommittee.
+
+All full nodes including RPCs should halt as soon as a safety
+violation is detected.
+
+### Subcommittee Rotation
+
+Voting Subcommittee is scheduled for a voting epoch, and is itself
+composed of 2 subcommittees, primary and secondary, scheduled for
+two epochs each in a staggered rotation.
+
+```
+A1 A1 A2 A2 A3
+   B1 B1 B2 B2
+```
+
+The first epoch the subcommittee runs, it is considered **secondary**.
+On the second epoch, it is considered **primary**.
+
+For a new secondary subcommittee to start, it must start with a
+block that is a descendant of a **rotation block** that was confirmed
+by both super-majorities in the previous epoch. The **rotation seed**
+is used to derive the new secondary subcommittee.
+
+### Subcommittee Selection
+
+A stake weighted shuffle from the **rotation seed** is used to pick
+the next secondary subcommittee. A slow hash is necessary to ensure
+that grinding the **rotation block** bankhash is not practical. A
+random sample of 200 nodes picked with a stake weighed shuffle.
+Assuming 1/3 are faulty, probability of 133 or greater faulty nodes
+is 1:10^20. A 1ms hash would require grinding for 10^9 years with
+1 billion cores.
+
+### Threshold Switching
+
+Secondary subcommittee can use the primary's votes as a switching
+threshold proof. The primary can only use its own votes for switching
+proofs.
+
+When secondary uses to switch forks using the primary votes, the
+fork that its switching from must be from the secondary epoch.
+
+```
+A1 A1 A2 A2 A3
+   B1 B1 B2 B2
+```
+
+Last block of epoch (A1,B1), X, A1 optimistically confirmed slot
+X. B1 since it was locked out did not. On the next slot, X+1, B1
+is now primary. B1 may use votes from A1 on slot X to switch forks.
+The fork form which B1 is switching from must be from an epoch when
+B1 was secondary.
+
+### Safety
+
+```
+A1 A1 A2 A2 A3
+   B1 B1 B2 B2
+```
+
+To switch subcommittees both currently active primary and secondary
+must confirm the same slot with a 2/3+ super-majority. When voting
+subcommittee switches from (A1,B1) to (A2,B1), B1 must have rooted
+the same fork as A1, and (A2,B1) must include the same fork, or a
+lockout violation has occurred, or the network has stalled.
+
+### Optimistic Confirmation
+
+A slot is optimistically confirmed if and only if both subcommittees
+confirm it with 2/3+ majority.
+
+Voting subcommittee (A1,B1) confirms the last slot `X` of its
+epoch, (A2,B1) take over on the same epoch. B1 must root `X`, or B1
+has committed a threshold switch violation. Since B1 is primary,
+it can only use its own votes for a switching proof.
```

Comment:
This seems to conflict with the statement above where B1 can also use A1's votes

"B1 may use votes from A1 on slot X to switch forks.
The fork form which B1 is switching from must be from an epoch when
B1 was secondary."

Reply:
if both confirm a slot, B1 cannot switch because A1 also confirmed, and therefore no fork has > 1/3 from either subcommittee.  you can't mix votes to create a proof.